Fix unitalized values returned from glx_attrib_list_from_attrib_list

mpjeff opened the pull request:
Fix for an issue causing me to get the following error in the examples

(lt-gears:1050): GdkGLExt-CRITICAL *_: gdk_x11_gl_config_new_for_screen: assertion `impl != NULL' failed
*_\* No appropriate OpenGL-capable visual found.

tdz commented:
Hi Jeff,

Thanks a lot for the fix! I should have noticed this by myself in the
first place.

I'd like to see a few changes before I put the changes into my tree. The
patch is actually three patches that
- rename the variable i to attrib_index,
- replace the terminating 0 by 'None', and
- fix the index bug by introducing glx_attrib_index.

I don't mind the changes, but would like to have them split into
separate patches; preferably in the order above. For the commit message
please stay within 78 characters for the headline, and 74 characters for
any detailed descriptions.

Thanks and best regards,
Thomas

Am Sonntag, den 01.07.2012, 12:43 -0700 schrieb jeff:

> Fix for an issue causing me to get the following error in the examples
> 
> (lt-gears:1050): GdkGLExt-CRITICAL *_: gdk_x11_gl_config_new_for_screen: assertion `impl != NULL' failed
> *_\* No appropriate OpenGL-capable visual found.
> 
> You can merge this Pull Request by running:
> 
>   git pull https://github.com/mpjeff/gtkglext master
> 
> Or you can view, comment on it, or merge it online at:
> 
>   https://github.com/tdz/gtkglext/pull/2
> 
> -- Commit Summary --
> - Fix skipped parameters leaving unitialized values in glx_attrib_list_from_attrib_list
> 
> -- File Changes --
> 
> M gdk/x11/gdkglconfig-x11.c (25)
> 
> -- Patch Links --
> 
>   https://github.com/tdz/gtkglext/pull/2.patch
>   https://github.com/tdz/gtkglext/pull/2.diff
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/tdz/gtkglext/pull/2

## 

GnuPG:          http://tdz.users.sourceforge.net/tdz.asc
Fingerprint:    16FF F599 82F8 E5AA 18C6 5220 D9DA D7D4 4EF1 DF08

jsapigen - A free glue-code generator for Mozilla SpiderMonkey. See
http://jsapigen.sourceforge.net for more information.

mpjeff committed:
Rename variable i in glx_attrib_list_from_attrib_list to attrib_index

mpjeff committed:
Use None at end of glx_attrib_list to match glXChooseVisual documentation

mpjeff committed:
Fix uninitialized values returned from glx_attrib_list_from_attrib_list

Skipped parameters GDK_GL_USE_GL and GDK_GL_RGBA caused the
corresponding spots in the returned array to not be initialized

mpjeff commented:
Thanks for the feedback!

I updated my tree based on your suggestions. Let me know if there is anything else.

tdz commented:
Hi Jeff

> I updated my tree based on your suggestions. Let me know if there is anything else.

Applied. Thanks a lot!

Best regards,
Thomas

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/tdz/gtkglext/pull/2#issuecomment-6725946

## 

GnuPG:          http://tdz.users.sourceforge.net/tdz.asc
Fingerprint:    16FF F599 82F8 E5AA 18C6 5220 D9DA D7D4 4EF1 DF08

jsapigen - A free glue-code generator for Mozilla SpiderMonkey. See
http://jsapigen.sourceforge.net for more information.
